src/coreclr/jit/ssabuilder.cpp

```diff
@@ -581,18 +659,40 @@ void SsaBuilder::AddPhiArg(
     // will be first in linear order as well.
     phi->gtUses = new (m_pCompiler, CMK_ASTNode) GenTreePhi::Use(phiArg, phi->gtUses);
 
-    GenTree* head = stmt->GetTreeList();
-    assert(head->OperIs(GT_PHI, GT_PHI_ARG));
-    stmt->SetTreeList(phiArg);
-    phiArg->gtNext = head;
-    head->gtPrev   = phiArg;
+#if defined(TARGET_WASM)
+    if (m_pCompiler->compRationalIRForm)
+    {
+        // TODO: multiple phis in one block
+        GenTree* firstNode = block->GetFirstLIRNode();
+        block->SetFirstLIRNode(phiArg);
+        phiArg->gtNext    = firstNode;
+        firstNode->gtPrev = phiArg;
+    }
+    else
+    {
+#endif
+        GenTree* head = stmt->GetTreeList();
+        assert(head->OperIs(GT_PHI, GT_PHI_ARG));
+        stmt->SetTreeList(phiArg);
+        phiArg->gtNext = head;
+        head->gtPrev   = phiArg;
+#if defined(TARGET_WASM)
+    }
+#endif
 
 #ifdef DEBUG
-    unsigned seqNum = 1;
-    for (GenTree* const node : stmt->TreeList())
+#if defined(TARGET_WASM)
+    if (!m_pCompiler->compRationalIRForm)
```

Comment:
```suggestion
    if (!block->IsLIR())
```
This double `#ifdef` is kind of ugly... Not clear how to make it better though, sadly.